graphql_compiler/tests/integration_tests/integration_test_helpers.py

```diff
@@ -88,10 +90,15 @@ def compile_and_run_sql_query(
     engine: Engine,
 ) -> Tuple[List[Dict[str, Any]], Dict[str, OutputMetadata]]:
     """Compile and run a SQL query against the SQL engine, return result and output metadata."""
-    compilation_result = graphql_to_sql(sql_schema_info, graphql_query, parameters)
+    compilation_result = compile_graphql_to_sql(sql_schema_info, graphql_query)
```

Comment:
I'm considering implementing the top-level `graphql_to_sql` function like this (print the query, wrap it in `text`, then attach parameters) so that we don't have 2 different execution paths to test.

Reply:
Are there any downsides we should be worried about?

Reply:
The performance downside is negligible.

I'm not aware of any semantic differences, but if there are any, we'd rather know sooner rather than later.

One other difference is that the resulting query would be more opaque, since it wouldn't be a sqlalchemy selectable, but a text query.

Reply:
I'm not really worried about performance. I'm worried about two things: the likelihood of running into bugs (security or correctness) in the insertion of arguments into a `text()` representation of the query, and the loss of flexibility with the generated query. When we return `Selectable`, the user is able to take that and use it in many different ways: add clauses (e.g. `LIMIT` or `ORDER BY`), plug it in elsewhere without immediately executing the query (e.g. use it to create a new vertex backed by the `Selectable`) etc.

Not having two different execution paths certainly simplifies _our_ lives as maintainers. But I'm not convinced it's the right thing to do from a user-facing perspective. So I'm inclined to leave the flexibility in and take the slight maintainability hit here.

What do you think?

Reply:
No strong opinions. Leaving it as is, and testing both execution paths against each other.